Adyen/ApiSerialization/SaleToPoiMessageSerializer.cs

```diff
@@ -64,6 +70,35 @@ public SaleToPOIResponse Deserialize(string saleToPoiMessageDto)
             return deserializedOutputMessage;
         }
 
+        public SaleToPOIRequest DeserializeNotification(string terminalNotificationJson)
```

Comment:
Suggestion to rewrite the following:

```csharp  
public SaleToPOIRequest DeserializeNotification(string terminalNotificationJson)
        {
            // Parse JsonObject
            var saleToPoiRequestJson = JObject.Parse(terminalNotificationJson);
            var saleToPoiRequestItem = saleToPoiRequestJson.FirstOrDefault();
            var saleToPoiRequestType = saleToPoiRequestItem.FirstOrDefault();
            string stringItem = saleToPoiRequestType?.ToString();

            // Deserialize Message Header
            var messageHeader = DeserializeMessageHeader(saleToPoiRequestType);

            // Get Payload and create new SaleToPOIRequest object
            if (stringItem.Contains("DisplayRequest"))
            {
                return new SaleToPOIRequest()
                {
                    MessageHeader = messageHeader,
                    MessagePayload = saleToPoiRequestType["DisplayRequest"]?.ToObject<DisplayRequest>()
                };
            }

            if (stringItem.Contains("EventNotification"))
            {
                return new SaleToPOIRequest()
                {
                    MessageHeader = messageHeader,
                    MessagePayload = saleToPoiRequestType["EventNotification"]?.ToObject<EventNotification>()
                };
            }

            throw new Exception("Json input is not a terminal notification.");
        }
```

Suggestion: rename `stringItem` to something more appropiate

Reply:
Guard clause approach: https://maximegel.medium.com/what-are-guard-clauses-and-how-to-use-them-350c8f1b6fd2